### PR TITLE
JAMES-3953 Performance enhancements

### PR DESCRIPTION
**Pull request by chibenwa**
Avoid partial reads and array concatenations.

CF: https://github.com/linagora/james-project/pull/5032

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

File: server/blob/blob-file/src/main/java/org/apache/james/blob/file/FileBlobStoreDAO.java [hunk 1]
```diff
@@ -97,13 +95,23 @@ public Mono<InputStream> readReactive(BucketName bucketName, BlobId blobId) {
...
+        }).onErrorResume(NoSuchFileException.class, e -> Mono.error(new ObjectNotFoundException(String.format("Cannot locate %s within %s", blobId.asString(), bucketName.asString()), e)))
+            .subscribeOn(Schedulers.boundedElastic());
     }
 
     @Override
     public Mono<Void> save(BucketName bucketName, BlobId blobId, byte[] data) {
-        return save(bucketName, blobId, new ByteArrayInputStream(data));
+        Preconditions.checkNotNull(data);
+        File bucketRoot = getBucketRoot(bucketName);
+        File blob = new File(bucketRoot, blobId.asString());
```
Comment (chibenwa) [inline server/blob/blob-file/src/main/java/org/apache/james/blob/file/FileBlobStoreDAO.java:110]: Needs to be wrapped in a mono....

-> Beginner mistake!

**Comment by chibenwa** [comment]
@vttranlina I think the perf tests you did for choosing the default BlobStoreDAO are invalidated by this change.

Don't you mind bench again this VS PGBlobStore?

**Comment by vttranlina** [comment]
> @vttranlina I think the perf tests you did for choosing the default BlobStoreDAO are invalidated by this change.
> 
> Don't you mind bench again this VS PGBlobStore?

postgresVsFile
The gap between min and max of Postgres is very high  
postgres: https://vttranlina.github.io/gatling_20240117/postgres/index.html
file: https://vttranlina.github.io/gatling_20240117/file/index.html

My screnario
```
export DURATION=7
export USER_COUNT=3000
export MAX_DURATION=16
```

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1906/8/

Related or not? I've seen lots of instability on those tests around those changes for the past few days

**Pull request merged**